### DISCUSSION TITLE
Twitter API updates

### DISCUSSION THREAD

**Discussion by pranavrajs** [chatwoot/chatwoot#6983] [General]
Hello everyone,

We wanted to update you on a recent impact made on the [Twitter inbox feature](https://www.chatwoot.com/features/twitter-integration) within Chatwoot.

So, Twitter has made significant changes to its APIs recently. This includes limitations on the number of requests raised per minute, plus substantial pricing changes. This has forced us to discontinue the Twitter integration temporarily. This means you won't be able to connect your Twitter inbox with Chatwoot and manage your DMs and Tweets from the dashboard.

We are very sorry for any inconvenience this may cause and understand that this may hinder your workflow. This change is as sudden and unexpected for us as for you. 

Please be assured that our team is actively working on quickly rolling out a new integration option and will keep you updated on any developments. In the meantime, we encourage you to contact us if you have any questions or concerns about this change. You can also [join our community on Discord](https://discord.com/invite/cJXdrwS).

### Answering a few questions you might have

**What happens to my active Twitter conversations?**

Conversations that were created before the integration was discontinued will remain in Chatwoot. However, you will not be able to reply to these conversations or receive any new updates for them.

**Does this impact other inboxes?**
No, all the other inboxes and Chatwoot features (as subscribed by you) will work just fine. 

**Will you bring back the Twitter integration?**
Certainly, the new integration options would look different, but we will try to be as inclusive as possible.

**Comment by Haashimi** [comment]
@Pranav Raj S
are still twitter integration discntinue

**Reply by pranavrajs** [reply]
We have discontinued the integration until further notice.

**Comment by mrspir** [comment]
If a developer can still create a free API without too many requests, why remove this feature? Should it be brought back in the Self Host version?

**Reply by sojan-official** [reply]
We need Account [Activity API access](https://developer.twitter.com/en/docs/twitter-api/enterprise/account-activity-api/overview) to work in real time. This is only available in the Enterprise plan, not available in the $100 plan. Unless it changes, we will not consider adding the channel back.

**Comment by tobsowo** [comment]
Well, another option will be to allow adding API credentials per workspaces or inbox and not the global option. That way, each workspace work with their own Developer access and good that getting developer access is quite easy with Twitter.

**Reply by mrspir** [reply]
I also agree with this. I've seen them update the new API price for the Basic package. If used with low capacity for customer support, $100 is quite suitable. We can bring this feature back

**Reply by pranavrajs** [reply]
We need Account [Activity API access](https://developer.twitter.com/en/docs/twitter-api/enterprise/account-activity-api/overview) to work in real time. This is only available in the Enterprise plan, not available in the $100 plan. Unless it changes, we will not consider adding the channel back.

**Comment by oleg-rswll** [comment]
Are there any updates on possible solutions for this?

**Reply by sojan-official** [reply]
https://github.com/orgs/chatwoot/discussions/6983#discussioncomment-7236091

**Comment by erikcik** [comment]
since the new updates on the api plans, is there any plans to bring back twitter?

**Reply by sojan-official** [reply]
we will be revisiting this once the api is generally available / we get the access

**Reply by khalid244** [reply]
any update about twitter ?
@sojan-official
